### ISSUE TITLE
[BUG] Harvester RKE2 cluster provisioning fails with air-gapped Rancher

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#1247]
**Describe the bug**

When Rancher is deployed in a network environment that requires HTTP proxy for internet access, provisioning Harvester RKE2 cluster fails because the machine-provisioner job fails to download the harvester driver from "releases.rancher.com".

**To Reproduce**
- Install Rancher in a network requirement that requires proxy for internet access.
- Specify proxy settings during Rancher installation (Helm variables)
- Create a RKE2 Harvester Cluster

**Expected behavior**
Rancher proxy settings are propagated to RKE2 machine provisioners for Harvester driver download from internet.

Harvester 0.3-pre
Rancher 2.6.1-RC1

**Comment by guangbochen** [comment]
will need to pass the installer proxy config to the rancher/rancherd,  @gitlawr can u PTAL of this, thanks

**Comment by gitlawr** [comment]
Need to be addressed in the upstream https://github.com/rancher/rancher/issues/34801

**Comment by guangbochen** [comment]
We will need to bump Rancher to the latest release once it was out and to verify the https://github.com/rancher/rancher/pull/34827 fix of it.

**Comment by janeczku** [comment]
Rancher 2.6.1 / Harvester 0.3.0

<img width="1401" alt="Screenshot 2021-10-14 at 23 39 27" src="https://user-images.githubusercontent.com/3813921/137399390-4cedf31f-5bc8-430c-a471-8264da5baeae.png">

**Comment by gitlawr** [comment]
@janeczku It works for me per https://github.com/rancher/rancher/issues/34801#issuecomment-949129183, can you help confirm?

**Comment by guangbochen** [comment]
@TachunLin @lanfon72 can one of u guys help to validate this, thanks.

**Comment by TachunLin** [comment]
@guangbochen , @gitlawr 
For better testing, may I know if we have an existing airgapped environment with proxy server in our data center to validate this issue or we can leverage such environment from rancher team.

**Comment by TachunLin** [comment]
Verified **fixed** on harvester 0.3.0 **master-119c2c48-head** with **Rancher 2.6.2**. Close this issue. 

### Result
Now we can provision RKE2 cluster from airgapped Rancher
![image](https://user-images.githubusercontent.com/29251855/142180267-35704fc1-542e-4e45-9ebd-ce5ba787e709.png)

### Environment Setup

* iPXE and DHCP server setup
  - Install an one node harvester with vagrant-pxe-harvester
  - ipxe example will prepare the following virtual network
    1. "harvester" (192.168.0.0/24, NAT)
    2. "vagrant-libvirt" (192.168.121.0/24, NAT)
* HTTP proxy server setup
  - Run vagrant ssh pxe_server to ssh into pxe server 
  - Install squid http proxy server `$ apt-get install squid`
  - edit /etc/squid/squid.conf and add line
    ```
    http_access allow all 
    http_port 3128
    ```
  - Restart squid service
  
* Install virtual machine and Rancher 2.6.2
  - Create an ubuntu 20.04 VM to install Rancher 
  - Set "harvester" and "vagrant-libvirt" virtual network
  - Install docker CE
  - Pull rancher 2.6.2 image
  - Get default route  `$ ip r | grep default`
  - delete default route `$ ip r delete <default route>  `
  - Create docker 2.6.2 container with http proxy setting:
    ```
      sudo docker run -d --restart=unless-stopped -p 80:80 -p 443:443 \
    -e HTTP_PROXY="http://192.168.0.254:3128" \
    -e HTTPS_PROXY="http://192.168.0.254:3128" \
    -e NO_PROXY="localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,192.168.0.0/16,.svc,.cluster.local,example.com" \
    --privileged rancher/rancher:v2.6.2
    ```
  
* Rancher 2.6.2 configuration
  - First time login rancher, set the url from https://localhost to  https://192.168.122.244 

* Harvester setup
  - Create a new virtual machine with cpu 12core, memory: 32GB 
  - Set "harvester" and "vagrant-libvirt" virtual network 
  - During ISO install console UI, select the NIC of "vagrant-libvirt" , to make harvester and rancher under the same subnet
  - After installation, enable  VLAN network with harvester-mgmt
  - Create new vlan with name: vlan1  and id: 1 
  - Create image and a new virtual machine by using vlan 1 network 
  - Confirm can retrieve ip address correctly 

### Test Information
* Test Environment: one node harvester on local kvm/qemu
* Rancher version: 2.6.2
* HTTP proxy server: squid
* Harvester version: master-119c2c48-head

### Verify Steps

1.  Import harvester from Rancher 
2.  Enable harvester node driver 
3.  Create harvester cloud credential 
4.  Create RKE2 cluster, given the following value in field  

![image](https://user-images.githubusercontent.com/29251855/142183675-ae73a27a-ce40-40cc-a2c8-1833e9bec1cc.png)

5. Check if there is a RKE2 cluster create correctly in running status and get ip address 

6. Wait for process to finish, (it will take 15 ~ 30 minutes)

### Additional Context
During the verification we found an issue: #1541 
If harvester installed by vagrant pxe example, when we enable vlan network and assign `bond0`
the whole harvester network will break, we cannot access harvester anymore
